### DISCUSSION TITLE
Website jumps when StackBlitz demo loads

### DISCUSSION THREAD

**Discussion by henryStelle** [pmndrs/jotai#2791] [Bug report] [answered]
### Bug Description

Reproduction steps:
1. Visit any page on https://jotai.org/docs with a StackBlitz demo
2. Wait a few seconds (as you might while reading the first paragraph)
3. Have the page suddenly scroll down to the demo

Example:
https://github.com/user-attachments/assets/dc22e779-7bd9-4f7b-8785-d8fd7c47df5f

### Reproduction Link

https://jotai.org/docs/utilities/storage

**Comment by dai-shi** [comment]
Thanks for reporting. I wonder if anyone can look into it.
cc: @niklasbec

**Reply by Brokyeom** [reply]
@dai-shi Hi, May I fix this bug?

**Reply by Brokyeom** [reply]
To address this issue, I initially added `tabIndex={-1}` to the StackBlitz iframe, but this didn’t resolve the problem.
Upon further inspection, the scroll jump seems to occur when the `npm install && npm run dev` command is executed in StackBlitz’s internal terminal.
It appears we may need a way to prevent focus from being triggered at the moment when interactions occur within the terminal.

**Reply by rothsandro** [reply]
It's a bug in Stackblitz but doesn't seem like they're going to fix it https://github.com/stackblitz/core/issues/2047. It works fine in Safari because there the iframe is lazy loaded but Chrome loads the iframe too early.

- We could add `&ctl=1` which shows a "Run project" button. This should fix the problem but the user always needs to confirm running the project.
- We could manually add lazy loading via intersection observer.
- We could try to prevent focusing the iframe (e.g. while it's not in viewport) but it's likely more complex and not sure if this can be done in a reliable, accessible way.

**Reply by dai-shi** [reply]
Thanks @rothsandro ! I don't think we should complicate things for this, so `&ctrl=1` is only a viable option?

**Reply by rothsandro** [reply]
@Brokyeom Do you want to create a PR?

**Reply by Brokyeom** [reply]
Thank you. I’ll give it a try.

**Reply by Brokyeom** [reply]
@dai-shi @rothsandro  I’ve created a PR; it’s #2798.

**Reply by dai-shi** [reply]
Thanks! Can @rothsandro review it please?

**Reply by niklasbec** [reply]
Saw this late, thanks for looking into it @Brokyeom

**Reply by rothsandro** [reply, marked as answer]
@Brokyeom Thanks for the PR, looks good 👍

**Reply by dai-shi** [reply]
Thanks all!
